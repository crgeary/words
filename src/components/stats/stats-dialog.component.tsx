import { PresentationChartBarIcon } from "@heroicons/react/outline";
import { FC } from "react";
import { Dialog } from "../dialog/dialog.component";
import { Stats } from "./stats.component";

export type StatsDialogProps = {
    isOpen: boolean;
    onClose: () => void;
};

export const StatsDialog: FC<StatsDialogProps> = ({ isOpen, onClose }) => {
    return (
        <Dialog
            isOpen={isOpen}
            onClose={onClose}
            heading="Statistics"
            icon={<PresentationChartBarIcon className="w-5 h-5 text-violet-400" />}
        >
            <Stats />
        </Dialog>
    );
};
