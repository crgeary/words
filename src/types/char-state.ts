export enum CharState {
    Correct = "correct",
    Present = "present",
    Absent = "absent",
    Idle = "idle",
}
