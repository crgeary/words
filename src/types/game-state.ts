export enum GameState {
    Idle = "idle",
    Playing = "playing",
    Fail = "fail",
    Success = "success",
}
