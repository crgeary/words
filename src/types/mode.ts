export enum Mode {
    Random = "random",
    Wordle = "wordle",
}

export type ModeConfig = {
    [k in Mode]: { name: string };
};
