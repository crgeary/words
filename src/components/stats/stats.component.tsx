import { FC } from "react";
import { StatsEntry } from "../../types/stats";
import NoDataImage from "./images/no-data.undraw.svg";

export type StatsProps = {
    items?: StatsEntry[];
};

export const Stats: FC<StatsProps> = ({ items = [] }) => {
    return (
        <div>
            {Array.isArray(items) && items.length === 0 ? (
                <div className="border border-dashed rounded p-8 flex justify-center items-center">
                    <div className="text-sm text-center">
                        <p className="mb-2 font-semibold">You haven&apos;t won any games!</p>
                        <p className="mb-8">To see your stats, go back and complete a game.</p>
                        <NoDataImage className="w-48 mx-auto" />
                    </div>
                </div>
            ) : (
                <p>Todo...</p>
            )}
        </div>
    );
};
