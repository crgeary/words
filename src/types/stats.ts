import { Mode } from "./mode";

export type StatsEntry = {
    mode: Mode;
    solution: string;
    guesses: string[];
    date: Date;
};
