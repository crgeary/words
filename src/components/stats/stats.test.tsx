import { render } from "@testing-library/react";
import { Mode } from "../../types/mode";

import { Stats } from "./stats.component";

test("it should match snapshot with no stats", () => {
    const { container } = render(<Stats items={[]} />);
    expect(container.firstChild).toMatchSnapshot();
});

test("it should match snapshot with stats", () => {
    const { container } = render(
        <Stats
            items={[
                {
                    guesses: ["hello", "world"],
                    solution: "world",
                    date: new Date(2022, 4, 14, 0, 0, 0, 0),
                    mode: Mode.Random,
                },
            ]}
        />,
    );
    expect(container.firstChild).toMatchSnapshot();
});
