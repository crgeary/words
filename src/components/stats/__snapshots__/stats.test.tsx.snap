// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should match snapshot with no stats 1`] = `
<div>
  <div
    class="border border-dashed rounded p-8 flex justify-center items-center"
  >
    <div
      class="text-sm text-center"
    >
      <p
        class="mb-2 font-semibold"
      >
        You haven't won any games!
      </p>
      <p
        class="mb-8"
      >
        To see your stats, go back and complete a game.
      </p>
      <span
        class="w-48 mx-auto"
        data-name="svgr-mock"
      />
    </div>
  </div>
</div>
`;

exports[`it should match snapshot with stats 1`] = `
<div>
  <p>
    Todo...
  </p>
</div>
`;
